src/main/java/in/twizmwaz/cardinal/module/modules/kit/KitCollection.java

```diff
@@ -0,0 +1,101 @@
+package in.twizmwaz.cardinal.module.modules.kit;
+
+import in.twizmwaz.cardinal.GameHandler;
+import in.twizmwaz.cardinal.event.CycleCompleteEvent;
+import in.twizmwaz.cardinal.module.modules.filter.FilterModule;
+import in.twizmwaz.cardinal.module.modules.filter.FilterState;
+import org.bukkit.Material;
+import org.bukkit.entity.EnderPearl;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.HandlerList;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class KitCollection implements KitRemovable {
```

Comment:
I do keep a refernce to the parent kits, the issue is you may try to get a parent kit that didn't get created yet, it stores a string, and on cycle complete, it uses the string to find the parent / parents, and stores them in the parentKits list